### ISSUE TITLE
packed struct/union bit casting panics

### ISSUE THREAD

**Issue by leighmcculloch** [ziglang/zig#14268]
### Zig Version

0.11.0-dev.1265+3ab43988c

### Steps to Reproduce and Observed Behavior

```
pub const Val = packed struct {
    const Self = @This();

    isTagged: bool,
    body: packed union {
        u63: u63,
        tagged: packed union {
            tags: enum(u3) {
                u32 = 0,
                i32 = 1,
                static = 2,
                object = 3,
                symbol = 4,
                bitSet = 5,
                status = 6,
            },
            body: packed union {
                static: enum(u60) {
                    void = 0,
                    true = 1,
                    false = 2,
                },
            },
        },
    },
};

const v: Val = @bitCast(Val, @as(u64, 2));
```

```
thread 2478416 panic: TODO implement readFromPackedMemory for more types
```

### Expected Behavior

I expected to be able to bit cast from a u64 to a Val, which is a 64-bit sized packed struct.

**Comment by nektro** [comment]
```
/home/meghan/src/zig/src/value.zig:1624:21: 0x1629fa8 in readFromPackedMemory (zig)
            else => @panic("TODO implement readFromPackedMemory for more types"),
                    ^
/home/meghan/src/zig/src/value.zig:1618:100: 0x1629883 in readFromPackedMemory (zig)
                        field_vals[i] = try readFromPackedMemory(field.ty, mod, buffer, bit_offset + bits, arena);
                                                                                                   ^
/home/meghan/src/zig/src/value.zig:1522:64: 0x1311432 in readFromMemory (zig)
                    return readFromPackedMemory(ty, mod, buffer[0..byte_count], 0, arena);
                                                               ^
/home/meghan/src/zig/src/Sema.zig:27190:80: 0x1485aa3 in bitCastVal (zig)
    return Value.readFromMemory(new_ty, sema.mod, buffer[buffer_offset..], sema.arena);
                                                                               ^
/home/meghan/src/zig/src/Sema.zig:27165:47: 0x1077890 in bitCast (zig)
        const result_val = try sema.bitCastVal(block, inst_src, val, old_ty, dest_ty, 0);
                                              ^
/home/meghan/src/zig/src/Sema.zig:9540:24: 0xf8552d in zirBitcast (zig)
    return sema.bitCast(block, dest_ty, operand, operand_src);
                       ^
/home/meghan/src/zig/src/Sema.zig:905:65: 0xd47a6b in analyzeBodyInner (zig)
            .bitcast                      => try sema.zirBitcast(block, inst),
                                                                ^
```

**Comment by perillo** [comment]
@Vexu: it seems that this issue has been fixed.  Tested with `0.12.0-dev.2341+92211135`.

**Comment by Vexu** [comment]
Fixed by #13124 which added test coverage.
